crates/router/src/core/payments/payment_methods.rs

```diff
@@ -131,7 +227,104 @@ struct RequiredFieldsAndSurchargeForEnabledPaymentMethodTypes(
     Vec<RequiredFieldsAndSurchargeForEnabledPaymentMethodType>,
 );
 
+fn get_extra_info_from_state(
+    state: &routes::SessionState,
+    payment_method_type: common_enums::enums::PaymentMethod,
+    payment_method_subtype: common_enums::enums::PaymentMethodType,
+    connectors: &Vec<api_models::enums::Connector>,
+) -> Option<api_models::payment_methods::PaymentMethodSubtypeSpecificData> {
+    match payment_method_type {
+        // TODO: Return card_networks
+        common_enums::PaymentMethod::Card | common_enums::PaymentMethod::CardRedirect => None,
+
+        common_enums::PaymentMethod::BankRedirect
+        | common_enums::PaymentMethod::BankTransfer
+        | common_enums::PaymentMethod::BankDebit => {
+            if let Some(connector_bank_names) =
+                state.conf.bank_config.0.get(&payment_method_subtype)
+            {
+                let mut bank_names = Vec::new();
+
+                for connector in connectors {
```

Comment:
can we use iteration based logic here instead of for loop and lets not take full state in the argument